### PR TITLE
[google-apps-script] avoid duplicated declaration for `console` object with @types/node

### PR DESCRIPTION
**Pull request by nokazn**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [ ] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [ ] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [ ] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: #32585
- ~~If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.~~

If removing a declaration:
- [ ] If a package was never on Definitely Typed, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### Details

If `@types/google-apps-script` is used with `@types/node` included in other dev dependencies, declaration for `console` object conflicts with the one in `@types/node`.
This PR resolves the above problems, I think.

### CODE REVIEW

File: types/google-apps-script/test/google-apps-script-tests.with-node.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
+// tslint:disable: no-var-keyword prefer-const object-literal-shorthand
```
[the remark this answers is not in the document]
Comment (nokazn) [reply types/google-apps-script/test/google-apps-script-tests.with-node.ts]: fixed by https://github.com/DefinitelyTyped/DefinitelyTyped/pull/60676/commits/362513a83cef79938ae637a2b7a1f3c9f2b27f6f.

File: types/google-apps-script/google-apps-script.base.d.ts [hunk 2]
```diff
@@ -6,6 +6,19 @@
 
 /// <reference path="google-apps-script.types.d.ts" />
 
+interface Console {
```
[the remark this answers is not in the document]
Comment (nokazn) [reply types/google-apps-script/google-apps-script.base.d.ts:9]: > User can't define Console type in top level. 

Yes, exactly.
But, type declarations of `console` object in [@types/node](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/cdceb7dbee6cfd9c9093ad0406792ff0f90472f7/types/node/console.d.ts) (or [lib.dom.d.ts](https://github.com/microsoft/TypeScript/blob/3bc9c80c1028c63642d3d3206de200a562500087/lib/lib.dom.d.ts#L17727-L17748)) are defined in global namespace. So, merging to `console` type with the one on @types/node needs to declare another `Console` type in global, I think. (`declare var console: GoogleAppsScript.Base.Console` conflicts with @types/node.)

[the remark this answers is not in the document]
Comment (mtgto) [reply types/google-apps-script/google-apps-script.base.d.ts:9]: > But, type declarations of console object in [@types/node](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/cdceb7dbee6cfd9c9093ad0406792ff0f90472f7/types/node/console.d.ts) (or [lib.dom.d.ts](https://github.com/microsoft/TypeScript/blob/3bc9c80c1028c63642d3d3206de200a562500087/lib/lib.dom.d.ts#L17727-L17748)) are defined in global namespace. So, merging to console type with the one in @types/node needs to declare another Console type in global, I think.

I see. Thank you for your explanation.

**Review by mtgto — APPROVED** [review]
I approve this pull request. You can merge this pull request. Thank you for your contribution!

At last, let me tell you my opinion.
My opinion is still "it is not good way to use both `@types/google-apps-script` and `@types/node` or `dom` library in one typescript project`.
The reason for this is Google Apps Script and Node.js are not SAME.

NOTE: I already wrote it: https://github.com/DefinitelyTyped/DefinitelyTyped/issues/32585#issuecomment-546343430 .

For example. This code can be transpiled, but it causes an error in Google Apps Script:

```ts
// NOTE: It causes an error:
// Exception: The parameters () don't match the method signature for console.timeEnd.
console.timeEnd();
```

In node.js, the argument is abbreviatable.
https://nodejs.org/api/console.html#consoletimeendlabel

But in Google Apps Script, the argument cannot be abberviate.
https://developers.google.com/apps-script/reference/base/console?hl=en#timeEnd(String)

This says Google Apps Script and Node.js has different interface.

Nevertheless, I understand that some people want to use `@types/google-apps-script` with `@types/node`, so I approve this.

**Comment by nokazn** [comment]
Ready to merge

**Pull request merged**